Sources/SearchableCollectionViewController/SearchableCollectionController.swift

```diff
@@ -1,51 +1,100 @@
+        controller.hidesNavigationBarDuringPresentation = true
+        controller.dimsBackgroundDuringPresentation = false
+
+        controller.searchBar.autoresizingMask = [.flexibleWidth]
+
+        return controller
+    }()
+
+    lazy var searchBarContainerView: UIView = {
+        let view = UIView(withAutoLayout: true)
+        view.backgroundColor = .clear
+
         return view
     }()
-    
-    var searchBackgroundColor: UIColor? = SearchBarView.defaultSearchBackgroundColor {
-        didSet { updateSearchColor() }
-    }
-    
-    override public func viewDidLoad() {
+
+    open override func viewDidLoad() {
         super.viewDidLoad()
-        
-        navigationController?.navigationBar.isTranslucent = false
-        view.addSubview(collectionView)
-        collectionView.fillSuperview()
+
+        self.automaticallyAdjustsScrollViewInsets = true
+
+        self.view.addSubview(self.searchBarContainerView)
+        self.searchBarContainerView.addSubview(self.searchBar)
+
+        self.searchBarContainerView.attachToTop(viewController: self)
+        self.searchBarContainerView.set(height: self.searchBar.frame.height)
+
+        self.searchBar.sizeToFit()
+
+        self.definesPresentationContext = true
+
+        self.collectionView.delegate = self
+        self.collectionView.dataSource = self
     }
-    
-    override public func viewWillAppear(_ animated: Bool) {
+
+    open override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        
-        updateSearchColor()
+
+        self.collectionView.contentInset.top += self.searchBar.frame.height
     }
-    
-    private func updateSearchColor() {
-        SearchBarView.appearance().searchBackgroundColor = searchBackgroundColor
+
+    open override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
     }
 }
 
-extension SearchableCollectionController: UICollectionViewDelegate {
-    public func scrollViewDidEndDragging(_ scrollView: UIScrollView, willDecelerate decelerate: Bool) {
-        if scrollView.contentOffset.y >= SearchBarView.height / 2 && scrollView.contentOffset.y < SearchBarView.height {
-            scrollView.setContentOffset(CGPoint(x: 0.0, y: SearchBarView.height), animated: true)
-        } else if scrollView.contentOffset.y > 0 && scrollView.contentOffset.y < SearchBarView.height / 2 {
-            scrollView.setContentOffset(CGPoint(x: 0.0, y: 0.0), animated: true)
+extension SearchableCollectionController: UICollectionViewDataSource {
+    open func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        return UICollectionViewCell()
+    }
+
+    open func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return 0
+    }
+}
+
+extension SearchableCollectionController: UICollectionViewDelegate, UIScrollViewDelegate {
```

Comment:
I think if I don't declare it here, I have to declare it for subclasses. Or is that just wrong? 🤔 Gonna check now!